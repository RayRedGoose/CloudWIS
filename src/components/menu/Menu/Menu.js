import React, { Component } from 'react'
import './Menu.scss'
import MenuIcon from '../_icons/MenuIcon/MenuIcon'
import SideMenu from '../SideMenu/SideMenu'

class Menu extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isActive: false
    }
  }

  toggleActiveState = () => {
    this.setState({ isActive: !this.state.isActive })
  }

  render() {
    return (
      <header className="side-menu">
        {
          (this.state.isActive)
            ? <SideMenu
              {...this.props}
              toggleActiveState={this.toggleActiveState}/>
            : <MenuIcon
              toggleActiveState={this.toggleActiveState} />
        }
      </header>
    )
  }
}

export default Menu
