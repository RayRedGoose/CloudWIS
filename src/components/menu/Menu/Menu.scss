.side-menu {
  position: absolute;
  top: 0; left: 0;
  z-index: 200;
  main {
    display: flex;
    height: 100vh; width: 100vw;
  }
}
