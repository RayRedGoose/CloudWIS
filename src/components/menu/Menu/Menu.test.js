import React from 'react'
import { shallow } from 'enzyme'
import Menu from './Menu'

describe('Menu', () => {
  let menu, mockProps

  beforeEach(() => {
    mockProps = {
      searchHistory: ['Denver', 'Phoenix'],
      changeCity: jest.fn(),
      fetchWeatherData: jest.fn()
    }

    menu = shallow(<Menu {...mockProps} />);
  })

  it('should match the snapshot if it\'s not clicked', () => {
    expect(menu).toMatchSnapshot();
  })

  it('should match the snapshot if it\'s clicked', () => {
    menu.setState({ isActive: true });

    expect(menu).toMatchSnapshot();
  })

  it('should change inactive state to active when toggleActiveState is called', () => {
    expect(menu.state('isActive')).toEqual(false);

    menu.instance().toggleActiveState();

    expect(menu.state('isActive')).toEqual(true);
  })

  it('should change active state to inactive when toggleActiveState is called', () => {
    menu.setState({ isActive: true });

    expect(menu.state('isActive')).toEqual(true);

    menu.instance().toggleActiveState();

    expect(menu.state('isActive')).toEqual(false);
  })

})
