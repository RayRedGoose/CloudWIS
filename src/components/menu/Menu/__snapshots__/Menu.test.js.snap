// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot if it's clicked 1`] = `
<header
  className="side-menu"
>
  <SideMenu
    changeCity={[MockFunction]}
    fetchWeatherData={[MockFunction]}
    searchHistory={
      Array [
        "Denver",
        "Phoenix",
      ]
    }
    toggleActiveState={[Function]}
  />
</header>
`;

exports[`Menu should match the snapshot if it's not clicked 1`] = `
<header
  className="side-menu"
>
  <MenuIcon
    toggleActiveState={[Function]}
  />
</header>
`;
